### PR TITLE
EVEREST-1274: check k8s version during install/upgrade

### PR DESCRIPTION
**Pull request by michal-kralik**
EVEREST-1274

### CODE REVIEW

File: pkg/install/install.go [hunk 1]
```diff
@@ -244,6 +253,57 @@ func (o *Install) Run(ctx context.Context) error {
...
+
+		if !supVer.kubernetes.Check(k8sVersion) {
+			return fmt.Errorf(
+				"kubernetes version %q does not meet minimum requirements of %q",
+				k8sVersion.String(), supVer.kubernetes.String(),
+			)
+		}
+
+		o.l.Debugf("Finished requirements check for Kubernetes version")
+	}
```
Comment (recharte) [inline pkg/install/install.go:302]: Can we use the `checkK8sRequirements()` function defined in the upgrade module, or vice-versa?

Comment (recharte) [reply pkg/install/install.go:302]: Just trying to avoid duplicate code :slightly_smiling_face:

Comment (michal-kralik) [reply pkg/install/install.go:302]: we could refactor, but it required extracting upgrade/install structures into a common package so I decided it's maybe better not to touch this behavior so we don't break it and make the PR smaller.

Comment (recharte) [reply pkg/install/install.go:302]: It's not a big deal but I think it's still work considering this refactor but it can be done in a follow-up PR.

**Review by recharte — APPROVED** [review]
Thanks Michal, it looks good. Let's consider doing the refactoring of the supported versions functions in a follow-up PR.

**Pull request merged**